### ISSUE TITLE
Animated connections w/o dashed lines have icons

### ISSUE THREAD

**Issue by danielsuh05** [terrastruct/d2#1949]
Got this from looking at issue #1461. 

Just some brainstorming, what would the syntax look like? Putting `animated: true` makes the line dashed automatically, so how would we differentiate whether the user wants a dashed line or a line with a moving icon? I'd like to take this issue if possible as well.

**Comment by alixander** [comment]
I think https://github.com/terrastruct/d2/issues/12 is a pre-req to this.

Once connections have icons, `animated: true` can animate the icon instead of dashed line.

**Comment by danielsuh05** [comment]
Sorry, a bit confused about what #12 is asking for. Does it mean add the ability for D2 to add an icon along a path? For example, there would be some style such that you could put a circle at any point along the path?

**Comment by alixander** [comment]
https://play.d2lang.com/?script=qlDQtVOotFKo5lJQUFZwyU8tVsjLL8nIzEvnUlDITM7Ps1LIKCkpKLbS1wfxivVKUouKEotLikqTS_SS83P1U4uLU_NKMhNzilWN3IwMjXUTcxKLcvWKy9K5arm4qmAml-cXZRdTw0hAAAAA__8%3D&

**Comment by bo-ku-ra** [comment]
@alixander
such as using a character instead of an icon.
https://play.d2lang.com/?script=qlDQtVOotFJQKkpNzFHISUxKzVFSqOZSUFBWcMlPLVbIyy_JyMxL51KAyFkpKD2a3q-koJyWX6SQmJeZm1iSmZ-noKGQWqGH4OuiqNXkquUCBAAA__8%3D&

**Comment by alixander** [comment]
Yeah I know a label can do that, but I think it'd be common for users to want a label AND an icon while animating. So I don't want to animate the label

**Comment by bo-ku-ra** [comment]
yes, so as follows

```d2
x -> y:  {
  # Does nothing
  label: "real label"
  style.animated: true;
  style.animation-label: "●" 
}
```

**Comment by alixander** [comment]
Why introduce a new keyword? Adding a third keyword here means you expect someone to have a label, icon, AND an animated text. That's bloated
